src/pykeen/training/training_loop.py

```diff
@@ -415,6 +417,90 @@ def train(
 
         return result
 
+    def _train_epoch(
+        self, 
+        batches: DataLoader, 
+        callbacks: MultiTrainingCallback, 
+        sub_batch_size: Optional[int],
+        label_smoothing: float,
+        slice_size: Optional[int],
+        epoch: int,
+        only_size_probing: bool,
+    ) -> float:
+        """
+        Run one epoch.
+        
+        :param batches:
+            the batches to process
+        :param callbacks:
+            the callbacks to apply (wrapped into a single object)
+        :param sub_batch_size:
+            the sub-batch size to use
+        :param label_smoothing:
+            the label-smoothing to apply
+        :param slice_size:
+            the optional slice size
+        :param epoch:
+            the current epoch (only used to forward to callbacks)
+        :param only_size_probing:
+            whether to stop after the second batch
+        
+        :return:
+            the epoch loss
+        """
+        # Accumulate loss over epoch
+        current_epoch_loss = 0.0
+
+        # Flag to check when to quit the size probing
+        evaluated_once = False
+
+        for batch in batches:
+            # apply callbacks before starting with batch
```

Comment:
shoudn't pre-batch get the batch?

Reply:
Right now, none of the callbacks needs it.

They are all defined with `**kwargs`, so future modifications should not break any typing.